test/RollbackManagerTimelockCompound.unit.t.sol

```diff
@@ -139,11 +139,201 @@ contract Execute is ExecuteBase, RollbackManagerTimelockCompoundTest {
       assertEq(_lastExecuteTransactionCalls[i].eta, _eta);
     }
   }
+
+  function testFuzz_ExecuteRollbackWithinGracePeriod(
+    address[2] memory _targetsFixed,
+    uint256[2] memory _valuesFixed,
+    bytes[2] memory _calldatasFixed,
+    string memory _description,
+    uint256 _timeOffset
+  ) external {
+    (address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas) =
+      toDynamicArrays(_targetsFixed, _valuesFixed, _calldatasFixed);
+
+    // Propose a rollback
+    vm.prank(admin);
+    uint256 rollbackId = rollbackManager.propose(_targets, _values, _calldatas, _description);
+
+    // Queue the rollback
+    vm.prank(guardian);
+    rollbackManager.queue(_targets, _values, _calldatas, _description);
+
+    // Bound time offset to be within grace period (after executable time but before grace period expires)
+    uint256 gracePeriod = targetTimelock.GRACE_PERIOD();
+    _timeOffset = bound(_timeOffset, _timelockDelay(), _timelockDelay() + gracePeriod - 1);
+
+    // Fast forward to executable time within grace period
+    vm.warp(block.timestamp + _timeOffset);
+
+    // Should be in Queued state (within grace period)
+    assertEq(uint8(rollbackManager.state(rollbackId)), uint8(IGovernor.ProposalState.Queued));
+
+    // Execute should succeed
+    vm.prank(guardian);
+    rollbackManager.execute(_targets, _values, _calldatas, _description);
+  }
+
+  function testFuzz_ExecuteRollbackAfterGracePeriod(
+    address[2] memory _targetsFixed,
+    uint256[2] memory _valuesFixed,
+    bytes[2] memory _calldatasFixed,
+    string memory _description,
+    uint256 _timeOffset
+  ) external {
+    (address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas) =
+      toDynamicArrays(_targetsFixed, _valuesFixed, _calldatasFixed);
+
+    // Propose a rollback
+    vm.prank(admin);
+    uint256 rollbackId = rollbackManager.propose(_targets, _values, _calldatas, _description);
+
+    // Queue the rollback
+    vm.prank(guardian);
+    rollbackManager.queue(_targets, _values, _calldatas, _description);
+
+    // Bound time offset to be after grace period expires
+    uint256 gracePeriod = targetTimelock.GRACE_PERIOD();
+    _timeOffset = bound(_timeOffset, _timelockDelay() + gracePeriod + 1, _timelockDelay() + gracePeriod + 30 days);
+
+    // Fast forward past grace period
+    vm.warp(block.timestamp + _timeOffset);
+
+    // Should be in Expired state (past grace period)
+    assertEq(uint8(rollbackManager.state(rollbackId)), uint8(IGovernor.ProposalState.Expired));
+
+    // Execute should revert because state is not Queued
+    vm.prank(guardian);
+    vm.expectRevert(abi.encodeWithSelector(RollbackManager.RollbackManager__NotQueued.selector, rollbackId));
+    rollbackManager.execute(_targets, _values, _calldatas, _description);
+  }
+
+  function testFuzz_GracePeriodExpiryBoundary(
```

Comment:
The method under test here seems to be the `state` method, not the `execute` method, yet this is in the `Execute` test contract

Reply:
Also, why test the boundary explicitly. If we have a test for before, and one for after, make sure the exact GracePeriod test is included in the bound on whichever side we intend the functionality to operate.